src/deploy-appengine.ts

```diff
@@ -106,31 +104,25 @@ export async function run(): Promise<void> {
       await installGcloudSDK(gcloudVersion);
     }
 
-    // set PROJECT ID
-    if (projectId) {
-      await setProject(projectId);
-    } else if (serviceAccountKey) {
-      projectId = await setProjectWithKey(serviceAccountKey);
-    } else if (process.env.GCLOUD_PROJECT) {
-      await setProject(process.env.GCLOUD_PROJECT);
-    }
-    // Fail if no Project Id is provided if not already set.
-    const projectIdSet = await isProjectIdSet();
-    if (!projectIdSet) {
-      throw new Error(
-        'No project Id provided. Ensure you have either project_id and/or credentials inputs are set.',
-      );
-    }
-
-    // Either serviceAccountKey or GOOGLE_GHA_CREDS_PATH env var required
-    if (serviceAccountKey || process.env.GOOGLE_GHA_CREDS_PATH) {
-      await authenticateGcloudSDK(serviceAccountKey);
+    // Either credentials or GOOGLE_GHA_CREDS_PATH env var required
+    if (credentials || process.env.GOOGLE_GHA_CREDS_PATH) {
+      await authenticateGcloudSDK(credentials);
     }
     const authenticated = await isAuthenticated();
     if (!authenticated) {
       throw new Error('Error authenticating the Cloud SDK.');
     }
 
+    // set PROJECT ID
+    if (!projectId) {
+      if (credentials) {
```

Comment:
Just a thought for UX as I was scanning logs - Would it help to add an info log to show where the project ID was set from?

Reply:
Done